syntax = "proto3";

package message;

// > Server directory
// protoc --cpp_out=gen -I=. message.proto
// protoc --grpc_out=gen --plugin=protoc-gen-grpc=`which grpc_cpp_plugin` -I. message.proto

service EmailVerifyService {
    rpc getEmailVerifyCode(EmailVerifyRequest) returns (EmailVerifyResponse);
}

message EmailVerifyRequest {
    string email = 1;
}

message EmailVerifyResponse{
    string status = 1;
    string message = 2; // error message
}


//////////////////////////////////////////////////////////////////////////////////
message ChatServerRequest {
    uint32 id = 1;      // 数据库中的用户 id
}

message ChatServerResponse {
    string host = 1;
    string port = 2;
    string token = 3;   // 登录到 ChatServer 的 token
    string message = 4; // error message
}


message LoginRequest {
    uint32 id = 1;
    string token = 2; // 登录到 ChatServer 的 token
}

message LoginResponse {
    string message = 1;
}

service StatusService {
    // 客户端 获取 ChatServer 的信息
    rpc getChatServerInfo (ChatServerRequest) returns (ChatServerResponse);

    // ChatServer 验证 客户端的 登录token
    rpc verifyToken (LoginRequest) returns (LoginResponse);
}